services/web/server/src/simcore_service_webserver/projects/projects_utils.py

```diff
@@ -168,3 +170,18 @@ async def project_uses_available_services(
     }
 
     return needed_services.issubset(available_services)
+
+
+async def project_get_depending_nodes(
+    project: Dict[str, Any], node_uuid: str
+) -> Set[str]:
+    depending_node_uuids = set()
+    for dep_node_uuid, dep_node_data in project.get("workbench", {}).items():
+        for dep_node_inputs_key_data in dep_node_data.get("inputs", {}).values():
+            if (
+                isinstance(dep_node_inputs_key_data, dict)
```

Comment:
is there any case in which this is not a ``dict``??

Reply:
yes. all values such as int, str, bool